relay-server/src/metrics_extraction/generic.rs

```diff
@@ -87,13 +87,24 @@ where
         });
     }
 
+    for mapping in &config.tags {
+        let mut tags_opt = None;
+
+        for metric in &mut metrics {
+            if mapping.matches(&metric.name) {
+                let tags = tags_opt.get_or_insert_with(|| extract_tags(instance, &mapping.tags));
```

Comment:
Are we doing the `get_or_insert_with` here to compute the tags lazily, in case no tag mapping matches? I'm not strongly opinionated, but I think extracting tags proactively could make this easier to read.

Reply:
That's the reason. There's a good chance that none of the extracted metrics matches, and in that case we won't compute the tags.
I'm happy to rewrite this for better readability, what if I call it `lazy_tags` instead?

Reply:
Sounds good to me!